app/src/main/java/com/rkm/tasky/util/image/ImageUtils.kt

```diff
@@ -0,0 +1,61 @@
+package com.rkm.tasky.util.image
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import androidx.core.net.toUri
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.IOException
+
+private const val IMAGE_THRESHOLD = 1024L * 1024L
+
+fun getImageFromUri(context: Context, uri: String): ByteArray? {
+    if(!doesImageExist(uri)) return null
+    var image: ByteArray ?= null
+    try {
+        image = getBytes(context, uri.toUri())
+    } catch (_: IOException) {
+        return null
+    }
+
+    if(getImageSize(uri) > IMAGE_THRESHOLD) {
+        image = image?.let { compressImage(it) }
+    }
+
+    return image
+}
```

Comment:
Here you have blocking, but non-suspending code, which can be quite dangerous in a coroutine environment. Nothing forces the caller of this function to execute it on the right thread (and not block the main thread). Most coroutines are launched on the main dispatcher by default which leads to asynchronous behavior as long as suspending functions are executed inside. But, if you execute blocking non-suspending functions, you'll freeze the UI thread!
Your code should instead follow the principle of main-safety, which means that every blocking function should be safe to call from the main thread. You achieve this pretty easily like this:
```
suspend fun blockingNonSuspendingCall() {
    withContext(Dispatchers.IO) { // Or Dispatchers.Default if it's CPU heavy work like image compression
    
    }
}
```  
That way, it doesn't matter if you call this function from a main dispatcher coroutine as it will make sure by itself that it runs on the right thread. 

For more information, watch this: https://www.youtube.com/watch?v=4_fDefmI3yI